File: Program's_Contributed_By_Contributors/C++ Programs/Sorting/bubblesort.cpp
#include<iostream>
#include<vector>
using namespace std;

void bubbleSort(vector<int>& arr, int n)
{   
    for(int i = 1; i<n; i++) {
        //for round 1 to n-1
        bool swapped = false;
        
        for(int j =0; j<n-i; j++) {
            
            //process element till n-i th index
            if(arr[j] > arr[j+1]) {
                swap(arr[j], arr[j+1]);
                swapped = true;
            }
            
        }
        
        if(swapped == false) {
            //already sorted
            break;
        }
           
    }

    cout<<"Your sorted array is: ";
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
}

int main(){

    int n;
    cout<<"Enter the size of array: ";
    cin>>n;
    vector<int> arr(n);
    for(int i=0; i<n; i++){
        cin>> arr[i];
    }
   
    bubbleSort(arr, n);
     
  return 0;
}

File: Program's_Contributed_By_Contributors/Golang Programs/bubble_sort.go
package main

import "fmt"

func main() {
    numbers := []int32{4, 2, 6, 8, 9, 4, 1}

    bubbleSort(numbers)

    fmt.Println(numbers)
}

func bubbleSort(arr []int32) {
    n := len(arr)
    swapped := true
    for swapped {
        swapped = false
        for i := 1; i < n; i++ {
            if arr[i-1] > arr[i] {
                arr[i-1], arr[i] = arr[i], arr[i-1]
                swapped = true
            }
        }
        n--
    }
}
